### ISSUE TITLE
One error for calculating the advantage value

### ISSUE THREAD

**Issue by GeminiLight** [jingw2/neural-combinatorial-optimization-rl#1]
Hi, Jingw2!

Thanks for your great reproduction of "neural-combinatorial-optimization-rl", which offers me help to understand and implement relevant algorithms. However, there is an error in the file `rl-tsp.ipynb` of your project and please fix it to make this repo more practical.

An error is made for the gradient of `b` is still retained when calculating the advantage which is an operation that requires no gradient actually.
```python
advantage = L - b.view(-1)
```

One possible solution:
```python
advantage = L - b.view(-1).detach()
```

![image](https://user-images.githubusercontent.com/49940241/143217522-aa8b5cef-6bb0-4b47-a876-cd7c60d45b65.png)

Best regards,
Gemini Light
